### ISSUE TITLE
Blocking calls must be done in the executor or a separate thread

### ISSUE THREAD

**Issue by Tezzlicious** [py-smart-gardena/hass-gardena-smart-system#113]
On the latest master on HASS 2022.5.5:

Logger: homeassistant.config_entries
Source: util/async_.py:161 
First occurred: 13:40:14 (1 occurrences) 
Last logged: 13:40:14

```
Error setting up entry for gardena_smart_system
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 335, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/gardena_smart_system/__init__.py", line 52, in async_setup_entry
    await gardena_system.start()
  File "/config/custom_components/gardena_smart_system/__init__.py", line 89, in start
    await self.smart_system.authenticate()
  File "/usr/local/lib/python3.9/site-packages/gardena/smart_system.py", line 72, in authenticate
    self.token = self.oauth_session.fetch_token(
  File "/usr/local/lib/python3.9/site-packages/requests_oauthlib/oauth2_session.py", line 341, in fetch_token
    r = self.request(
  File "/usr/local/lib/python3.9/site-packages/requests_oauthlib/oauth2_session.py", line 521, in request
    return super(OAuth2Session, self).request(
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 529, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 645, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 440, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.9/site-packages/urllib3/connectionpool.py", line 398, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 239, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.9/http/client.py", line 1285, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.9/http/client.py", line 1296, in _send_request
    self.putrequest(method, url, **skips)
  File "/usr/local/lib/python3.9/site-packages/urllib3/connection.py", line 219, in putrequest
    return _HTTPConnection.putrequest(self, method, url, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/util/async_.py", line 173, in protected_loop_func
    check_loop(func, strict=strict)
  File "/usr/src/homeassistant/homeassistant/util/async_.py", line 161, in check_loop
    raise RuntimeError(
RuntimeError: Blocking calls must be done in the executor or a separate thread; Use `await hass.async_add_executor_job()` at custom_components/gardena_smart_system/__init__.py, line 89: await self.smart_system.authenticate()

```

Note: Release 0.2.5b4 works.

**Comment by grm** [comment]
Should be fixed in 1.0.0b900

Please update and tell me if it is working (in the other case, please close the ticket ;)

**Comment by grm** [comment]
Hello,

I will close this ticket.
If the issue is still present do not hesitate to reopen it/create another one.

Grm
